pkg/cmd/server/origin/master.go

```diff
@@ -692,3 +693,26 @@ type clientDeploymentInterface struct {
 func (c clientDeploymentInterface) GetDeployment(ctx api.Context, name string) (*api.ReplicationController, error) {
 	return c.KubeClient.ReplicationControllers(api.NamespaceValue(ctx)).Get(name)
 }
+
+// namespacingFilter adds a filter that adds the namespace of the request to the context.  Not all requests will have namespaces,
+// but any that do will have the appropriate value added.
+func namespacingFilter(handler http.Handler, contextMapper kapi.RequestContextMapper) http.Handler {
+	infoResolver := &apiserver.APIRequestInfoResolver{util.NewStringSet("api", "osapi"), latest.RESTMapper}
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		ctx, ok := contextMapper.Get(req)
+		if !ok {
+			http.Error(w, "Unable to find request context", http.StatusInternalServerError)
+			return
+		}
+
+		if _, exists := kapi.NamespaceFrom(ctx); !exists {
+			requestInfo, _ := infoResolver.GetAPIRequestInfo(req)
+			ctx = kapi.WithNamespace(ctx, requestInfo.Namespace)
```

Comment:
An error doesn't necessarily mean that a namespace wasn't found.  From the godoc:
```
If error is not nil, APIRequestInfo holds the information as best it is known before the failure
```

An empty namespace is not the same as a missing namespace.  Empty namespace can mean any namespace, all namespaces, or no namespace depending upon context.  It is up to the namespace user to decide which is appropriate.